core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -152,260 +152,253 @@ public void destroyCluster() throws Exception {
-        KafkaShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(), "group1");
-        Set<String> subscription = Collections.singleton(tp.topic());
-        shareConsumer.subscribe(subscription);
-        assertEquals(subscription, shareConsumer.subscription());
         alterShareAutoOffsetReset("group1", "earliest");
-        ConsumerRecords<byte[], byte[]> records = shareConsumer.poll(Duration.ofMillis(500));
-        shareConsumer.subscribe(Collections.emptySet());
-        assertEquals(Collections.emptySet(), shareConsumer.subscription());
-        // "Consumer is not subscribed to any topics."
-        assertThrows(IllegalStateException.class, () -> shareConsumer.poll(Duration.ofMillis(500)));
-        shareConsumer.close();
-        assertEquals(0, records.count());
+        try (KafkaShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(), "group1")) {
+            Set<String> subscription = Collections.singleton(tp.topic());
+            shareConsumer.subscribe(subscription);
+            assertEquals(subscription, shareConsumer.subscription());
+            ConsumerRecords<byte[], byte[]> records = shareConsumer.poll(Duration.ofMillis(500));
+            shareConsumer.subscribe(Collections.emptySet());
+            assertEquals(Collections.emptySet(), shareConsumer.subscription());
+            // "Consumer is not subscribed to any topics."
+            assertThrows(IllegalStateException.class, () -> shareConsumer.poll(Duration.ofMillis(500)));
+            assertEquals(0, records.count());
+        }
     }
 
     @ParameterizedTest(name = "{displayName}.persister={0}")
     @ValueSource(strings = {NO_OP_PERSISTER, DEFAULT_STATE_PERSISTER})
     public void testSubscriptionAndPoll(String persister) {
-        ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(tp.topic(), tp.partition(), null, "key".getBytes(), "value".getBytes());
-        KafkaProducer<byte[], byte[]> producer = createProducer(new ByteArraySerializer(), new ByteArraySerializer());
-        producer.send(record);
-        KafkaShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(), "group1");
-        shareConsumer.subscribe(Collections.singleton(tp.topic()));
         alterShareAutoOffsetReset("group1", "earliest");
-        ConsumerRecords<byte[], byte[]> records = shareConsumer.poll(Duration.ofMillis(5000));
-        assertEquals(1, records.count());
-        shareConsumer.close();
-        producer.close();
+        try (KafkaProducer<byte[], byte[]> producer = createProducer(new ByteArraySerializer(), new ByteArraySerializer());
+             KafkaShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(), "group1")) {
```

Comment:
need to close shareConsumer here and other places ?

Reply:
Happens automatically because these resources are opened in a try-with-resource block. As the scope of the try block is exited, the resources are automatically closed.